src/Uno.Extensions.Core.Generators/PropertySelector/PropertySelectorCandidate.cs

```diff
@@ -12,32 +14,78 @@ namespace Uno.Extensions.Generators.PropertySelector;
+			!method.Parameters.Any(p => p.Type is INamedTypeSymbol
+			{
+				IsGenericType: true,
+				MetadataName: "PropertySelector`2",
+				ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
+			}))
 		{
-			IsGenericType: true,
-			MetadataName: "PropertySelector`2",
-			ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
-		});
-
-		Context = context;
-		Syntax = syntax;
-		Location = position;
-		Method = method;
-		IsValid = hasPropertySelector ?? false;
+			IsValid = false;
+			Selectors = null;
+			MethodGlobalNamespace = null;
+			MethodName = null;
+			return;
+		}
+
+		MethodName = method.Name;
+		MethodGlobalNamespace = method.ContainingModule?.GlobalNamespace?.ToString() ?? "";
+
+		var callerPathParameter = method.Parameters.FirstOrDefault(p => p.HasAttribute<CallerFilePathAttribute>());
+		var callerLineParameter = method.Parameters.FirstOrDefault(p => p.HasAttribute<CallerLineNumberAttribute>());
+
+		if (callerPathParameter is null || callerLineParameter is null)
+		{
+			IsValid = false;
+			Selectors = null;
+			return;
+		}
+
+		var arguments = Syntax.ArgumentList.Arguments;
+
+		var callerPathArg = arguments.ElementAtOrDefault(callerPathParameter.Ordinal)?.Expression;
+		var callerPath = callerPathArg switch
+		{
+			LiteralExpressionSyntax pathArg => pathArg.Token.ValueText,
+			// It might be something else, but in this case the analyzer will complain.
+			_ => Location.Path,
+		};
+
+		var callerLineArg = arguments.ElementAtOrDefault(callerLineParameter.Ordinal)?.Expression;
+		var callerLine = callerLineArg switch
+		{
+			LiteralExpressionSyntax lineArg => lineArg.Token.ValueText,
+			// It might be something else, but in this case the analyzer will complain.
+			_ => (Location.StartLinePosition.Line + 1).ToString(CultureInfo.InvariantCulture),
+		};
+
+		Selectors = method.Parameters
+			.Where(param => param.Type is INamedTypeSymbol
+			{
+				IsGenericType: true,
+				MetadataName: "PropertySelector`2",
+				ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
+			})
+			.Select(param => new PropertySelectorUsageItem(
+				(INamedTypeSymbol)param.Type,
+				arguments.ElementAtOrDefault(param.Ordinal),
+				$"{param.Name}{callerPath}{callerLine}"))
+			.ToImmutableList();
 	}
 
-	[MemberNotNullWhen(true, nameof(Syntax), nameof(Method))]
+	[MemberNotNullWhen(true, nameof(Syntax), nameof(Selectors), nameof(MethodGlobalNamespace), nameof(MethodName))]
 	public bool IsValid { get; }
 
-	public GeneratorSyntaxContext Context { get; }
-
 	public InvocationExpressionSyntax? Syntax { get; }
 	public FileLinePositionSpan Location { get; }
-
-	public IMethodSymbol? Method { get; }
+	public string? MethodGlobalNamespace { get; }
+	public string? MethodName { get; }
+	public ImmutableList<PropertySelectorUsageItem>? Selectors { get; }
```

Comment:
Note to self so that I don't forget. `ImmutableList` is not the correct type to use here. We need a custom equatable data structure.